bin/runtime/src/lib.rs

```diff
@@ -1306,6 +1319,173 @@ mod tests {
         }
     }
 
+    #[test]
+    // This test is to make sure that we don't break call-runtime.
+    fn test_staking_pallet_index() {
+        // arbitrary call that is easy to construct
+        let c = RuntimeCall::Staking(pallet_staking::Call::bond_extra { max_additional: 0 });
+        // first byte is pallet index
+        assert_eq!(c.encode()[0], 8);
+    }
+
+    #[test]
+    // This test is to make sure that we don't break call-runtime.
+    fn test_nomination_pools_pallet_index() {
+        // arbitrary call that is easy to construct
+        let c = RuntimeCall::NominationPools(pallet_nomination_pools::Call::chill { pool_id: 0 });
+        // first byte is pallet index
+        assert_eq!(c.encode()[0], 19);
+    }
+
+    fn match_staking_call(c: pallet_staking::Call<Runtime>) {
+        match c {
+            pallet_staking::Call::bond { value: _, payee: _ } => {}
```

Comment:
I would slightly insist on specifying types explicitly... I know that it might require whole litany of imports, but this would give us one more guarantee, imo an important one

Reply:
I have no idea how to do that, like in terms of labor, without spending infinite amount of time on it :P 

Hints?

Reply:
my IDE (I think it's actually rust-analyzer) hints with:
![image](https://github.com/Cardinal-Cryptography/aleph-node/assets/27450471/0fd56f78-fd86-4884-8632-6ee652e3a9ff)

unfortunately, I cannot automatically inline it...

Reply:
Ok, I tried. It's just too much manual work. Many of them require some specific imports, obviously each type has many places to import it from, there are aliases and shit like that. Also whenever I make an edit, my VSCode needs ~5sec to show the types again. It's just not worth it :/

Reply:
okay, let's give it up now, I will do it in a follow-up PR during some long meeting xD

Reply:
ok, sent on slack :)